### DISCUSSION TITLE
[Feature Request] load game update XML from URL

### DISCUSSION THREAD

**Discussion by shakeyourbunny** [MikeMaximus/gbm#233] [Ideas]
Hello,

it would be really neat if the GBM would be able to download custom XML update data automatically on startup / interval from a custom URL and integrate it to the database.

My guess these things would be necessary (along with default options):
[ ] enable savegame defintion XML updates from custom URL
[ .... the url .... ]
Update Interval (dropdown): on demand / on startup / daily / weekly / monthly
[x] display changes:
 (green) "+": new savegames
 (red) "-": removed entries
 (blue) "*": changed entries
[x] if there are new updates available, ask for confirmation before updating:  Accept / Ignore Updates
[ ] update and overwrite game definitions
[x] make a backup of definition database (with timestamp) in XML format.
[x] do not update automatically games with this tag [ #private ]

additional settings:
[x] on xml export, do not export entries marked with this tag [ #private ]

These options may live in an additional entry in the global settings dialog.
This would really make sharing custom XML savegame definitions A LOT easier.

**Comment by MikeMaximus** [comment]
I like the idea,  I'll look into adding a feature like this.

Thanks you for the suggestion, I haven't really had any feedback on users actually sharing their own custom lists.

**Reply by shakeyourbunny** [reply]
Well, I'll try to be polite, but frank and honest, after rewriting this answer for the third time, the success of GBM (in comparison with other ones and a certain long dead one) depends not only on the software itself, but also on the quality and volume of the savegame backup definitions.

Both are lacking and I want to host a community-sourced savegame definition XML on Github, but without direct download and update possibility this will not get off the ground due really really really clunky procedure to update and share the definition. With a repo, the entry for that is low and willingness to share and build such is way higher.

Oh, and rethink the "will only save the bare minimum for a save file". Try to have a heavy modded game with saves where only the bare savegame file is copied away. Have fun trying to restore that.
